inspirehep/modules/records/serializers/fields_export.py

```diff
@@ -0,0 +1,337 @@
+
+    Returns:
+        string: common, or if name ambiguity possible, unique name of a country
+
+    Note:
+        Countries in pycountry (with only official names, esp. historic_countries) will sometimes
+        be represented in a format like: 'Korea, Republic of'. Thus we reverse the order of segments around
+        the comma to make sound it natural: 'Republic of Korea'.
+
+        In particular, we cannot pick the first segment only due to ambiguities like
+        South/North Korea or East/West Congo
+    """
+    country_name_for_code = {country.alpha_2: ' '.join(reversed(country.name.split(', '))) \
+                             # Order below matters, 'last wins' in dictionary comprehension
+                             # (https://docs.python.org/3/reference/expressions.html#dictionary-displays)
+                             for country in list(pycountry.historic_countries) + list(pycountry.countries)}
+    return country_name_for_code.get(code, default)
+
+
+def get_date(data, doc_type):
+    publication_year = get_value(data, 'publication_info[0].year')
+    thesis_date = get_value(data, 'thesis_info.date')
+    imprint_date = get_value(data, 'imprints[0].date')
+
+    if doc_type.endswith('thesis'):
+        date_choice = thesis_date or publication_year or imprint_date
+    else:
+        date_choice = publication_year or thesis_date or imprint_date
+
+    if date_choice:
+        return PartialDate.parse(unicode(date_choice))
+
+
+# Functions below describe where the non-obvious data is located:
+#
+# Args:
+#   data (dict): data partially serialized by the means of defined marshmallow schema only
+#   doc_type (str): BibTeX document type, as reported by `bibtex_document_type`
+
+extractor = make_extractor()
+
+
+@extractor('author')
+def get_author(data, doc_type):
+    """
+    Note:
+        Only used to generate author field if corporate_author is the author.
+    """
+    if 'corporate_author' in data:
+        return ' and '.join('{{{}}}'.format(author) for author in data['corporate_author'])
+
+
+@extractor('journal')
+def get_journal(data, doc_type):
+    return get_value(data, 'publication_info[0].journal_title')
+
+
+@extractor('volume')
+def get_volume(data, doc_type):
+    publication_volume = get_value(data, 'publication_info[0].journal_volume')
+    bookseries_volume = get_value(data, 'book_series[0].volume')
+    return publication_volume or bookseries_volume
+
+
+@extractor('year')
+def get_year(data, doc_type):
+    date = get_date(data, doc_type)
+    if date:
+        return date.year
+
+
+@extractor('month')
+def get_month(data, doc_type):
+    date = get_date(data, doc_type)
+    if date:
+        return date.month
+
+
+@extractor('number')
+def get_number(data, doc_type):
+    return get_value(data, 'publication_info[0].journal_issue')
+
+
+@extractor('pages')
+def get_pages(data, doc_type):
+    return get_page_artid(data, '--')
+
+
+@extractor('primaryClass')
+def get_primary_class(data, doc_type):
```

Comment:
I would only do this for new-style archive identifiers, as the old-style ones have the category in their id already.